lib/controller_patches.rb

```diff
@@ -41,6 +36,54 @@ def frontpage
         @top_requests = InfoRequest.top_requests.limit(2)
       end
     end
+
+    def cached_blog
+      blog_cache("blog_posts-#{@locale}")
+      render :action => 'blog'
+    end
+
+    private
+
+    def blog_cache(cache_key, expires=4.hours)
+      # nothing to do here, call the blog code and return
+      return blog unless AlaveteliConfiguration::cache_fragments
+
+      # read in the last timestamp
+      updated = read_fragment("#{cache_key}-updated")
+
+      # construct the fragment key
+      @fragment_key = "#{cache_key}-#{updated}"
+
+      # if the fragment is unreadable or has reached expiry
+      # attempt to pull in the blog feed
+      if updated.nil? || !fragment_exist?(@fragment_key) ||
+         Time.now > (Time.parse(updated) + expires)
+        # keep a note of the old key
+        old_key = @fragment_key.dup
+
+        # prepare a new timestamp and fragment key
+        updated = Time.now.to_s
+        @fragment_key = "#{cache_key}-#{updated}"
+
+        logger.info "attempting to pull in the feed"
+        blog
+
+        if @blog_items.empty?
+          # attempt to write the previous fragment to our new cache
+          fragment = read_fragment(old_key)
+          if fragment
+            write_fragment(@fragment_key, fragment)
+            logger.warn "Writing old blog fragment (#{old_key}) to " \
+                        "current cache (#{@fragment_key}) due to feed " \
+                        "error or timeout."
+            write_fragment(@blog_updated_at, read_fragment(old_key))
```

Comment:
and I'm reading the fragment into a variable then reading it in again, should probably fix that as well